swarm/network/protocol_test.go

```diff
@@ -30,6 +30,11 @@ import (
 	p2ptest "github.com/ethereum/go-ethereum/p2p/testing"
 )
 
+const (
+	TestProtocolVersion   = 5
```

Comment:
The current version is 5. You are changing the handshake message, which means that you need to bump this to 6, no?

Reply:
I bumped it to 6 to reflect the fact that the handshake is changed. However I think changing here has little significance, because it is only used in the scope of these tests and it is only used in one test for comparing with zero. 

I think in order to change it in the protocol, one has to change the version in the `BzzSpec` that is inside `swarm/network/protocol.go`.

Also I remember in one of our hangout calls that it may not be necessary to bump this, @janos can you chime in?

Reply:
Well, I was wrong about the significance of changing this in the tests, because if I change it there and not in `BzzSpec` the tests will fail.

Still the question is valid, should we change it in `BzzSpec` and therefore here too?

Reply:
I ran some tests. 

If we change the version, then nodes running version 6 won't connect to nodes that are running version 5 (as expected). 

However if we just add the LightNode flag to the handshake without changing the version, the nodes are able to connect to each other. This is a bit surprising to me, but this means it's not strictly necessary to change the version at this point, because currently the code doesn't change any functionality.

In any case in the future it seems like a good idea to bump the version when add the actual LightNode functionality to Kademlia, because from that point nodes will definitely work differently, hence the need for the version change.

So I think the question is should we change the version two times (now and when the Kademlia changes are implemented) or just once (only after the Kademlia changes)?

Reply:
I think that adding a field to a message is not a breaking change for protocol. It just requires that default value should be handled, and since default for light node flag is false, then it is handled implicitly.

Reply:
OK, I thought nodes won't be able to deserialise the new value. If they can, then perfect!

Reply:
@janos That's what my experiment suggests too, that just adding a `LightNode` field to the `Handshake` message works even when connecting to nodes which do not support the `LightNode` flag.

So basically you suggest that we keep the version right now at 5 and change it to 6 when we actually introduce breaking change in the functionality in Kademlia? @nonsense wdyt?

Reply:
Ok, kept the version at 5 right now

Reply:
Apparently I don't understand exactly the RLP serialization - will play with this locally, and I trust you that it works. Since this is the case, keep it at 5, no need to upgrade.